ruff.toml

```diff
@@ -0,0 +1,16 @@
+line-length = 120
+indent-width = 4
+target-version = "py311"
+
+[lint]
+select = ["B", "C", "E", "F", "W", "I", "N", "Q", "UP", "RET", "RSE", "RUF", "ISC", "PLC", "PLE", "PLW", "T20", "PERF"]
+ignore = [
+    "B011",  # In AP, the use of assert False is essential because we optimise out these statements for release builds.
+    "C901",  # Author disagrees with limiting branch complexity
+    "N818",  # Author agrees with this rule, but Core AP violates this and changing it would be a hassle.
+    "PLC0415",  # In AP, we consider local imports totally fine & necessary
+    "PLC1802",  # Author agrees with this rule, but it literally changes the functionality of the code, which is unsafe.
+    "PLC1901",  # This is just not equivalent
+    "PLE1141",  # Ruff is not meant for error checking. If the code author meant .items(), they will realize soon enough
```

Comment:
I disagree with this comment.
Ruff is meant for error checking.
It's just not very good at checking this particular error. Type checkers are better for checking this particular error.

Reply:
(There are lots of other rules, especially in `"B"` that are more valuable for finding errors.)

Reply:
Well, seems we're at an impasse :D I definitely think ruff shouldn't try to fix the *behavior* of your code

Reply:
Obviously I have no delusions that we're gonna get there with AP, but still, I think the goal should always be 0 errors shown. Either by fixing them, or by disabling the rules you don't like

Reply:
I'm not sure how that would happen... I'm happy with how it is now.

Reply:
I changed the comment, is this more agreeable?